### PR TITLE
Add mention of XML support to RESTful binary sensor

### PR DESCRIPTION
**Pull request by 2sheds**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/110062
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -135,7 +135,7 @@ GEM
       rack-protection (= 4.0.0)
       rack-session (>= 2.0.0, < 3)
       tilt (~> 2.0)
-    sorbet-runtime (0.5.11367)
+    sorbet-runtime (0.5.11368)
```
Comment (c0ffeeca7) [inline Gemfile.lock:138]: please revert the changes to this file. They are unrelated to the PR.

**Comment by c0ffeeca7** [comment]
It seems a few changes sneaked into this PR that are unrelated to the XML support. Please remove those and make separate PRs for those changes. See the [guidelines on creating a PR](https://developers.home-assistant.io/docs/review-process#creating-the-perfect-pr).

**Comment by 2sheds** [comment]
Closing to replace with https://github.com/home-assistant/home-assistant.io/pull/33073
